src/psygnal/_signal.py

```diff
@@ -897,6 +908,13 @@ def emit(
 
             SignalInstance._debug_hook(EmissionInfo(self, args))
 
+        if queue:
+            with self._lock:
+                self._emit_queue.put(
+                    (args, {"queue": False, "asynchronous": asynchronous})
+                )
```

Comment:
Python Queues are thread safe (they are designed among others to thread communication). So using an additional thread looks obsolete.

Reply:
ok! thanks!